src/main/java/org/elmlang/intellijplugin/features/intention/imports/ElmAddImportHelper.java

```diff
@@ -0,0 +1,199 @@
+        Optional<ElmExposingClause> exposing2 = Optional.ofNullable(import2.getExposingClause());
+
+        List<String> exposedValues =
+                Stream.concat(
+                        exposing1.map(e -> e.getLowerCaseIdList().stream()).orElse(Stream.empty()),
+                        exposing2.map(e -> e.getLowerCaseIdList().stream()).orElse(Stream.empty())
+                )
+                .sorted((e1, e2) -> e1.getText().compareTo(e2.getText()))
+                .map(PsiElement::getText)
+                .collect(Collectors.toList());
+
+        Map<String,List<ElmExposedUnion>> exposedUnionsByName =
+                Stream.concat(
+                        exposing1.map(e -> e.getExposedUnionList().stream()).orElse(Stream.empty()),
+                        exposing2.map(e -> e.getExposedUnionList().stream()).orElse(Stream.empty())
+                )
+                .collect(
+                    Collectors.groupingBy(
+                        e -> e.getUpperCaseId().getName()
+                    )
+                );
+
+        List<String> exposedTypes = exposedUnionsByName.entrySet()
+                .stream()
+                .map(entry -> mergeExposedUnionConstructors(entry.getKey(), entry.getValue()))
+                .collect(Collectors.toList());
+
+        // combine the new exposing list and sort it
+        List<String> exposedNames = new ArrayList<>();
+        exposedNames.addAll(exposedValues);
+        exposedNames.addAll(exposedTypes);
+        exposedNames.sort(String::compareTo);
+
+        // generate the new, merged import statement
+        String moduleName = import1.getModuleName().getText();
+        String modulePlusAlias = moduleName + aliasText;
+        return ElmElementFactory.createImportExposing(project, modulePlusAlias, exposedNames);
+    }
+
+    private static String mergeExposedUnionConstructors(String typeName, List<ElmExposedUnion> unions) {
+        boolean exposingAll = unions.stream().anyMatch(union -> {
+            ElmExposedUnionConstructors ctors = union.getExposedUnionConstructors();
+            return ctors != null && ctors.isExposingAll();
+        }) ;
+
+        if (exposingAll) {
+            return typeName + "(..)";
+        }
+
+        String body;
+        body = unions.stream()
+                .filter(e -> e.getExposedUnionConstructors() != null)
+                .flatMap(e -> e.getExposedUnionConstructors().getUpperCaseIdList().stream())
+                .sorted((e1, e2) -> {
+                    String type1 = e1.getText();
+                    String type2 = e2.getText();
+                    return type1.compareTo(type2);
+                })
+                .map(e -> e.getName())
+                .collect(Collectors.joining(", "));
+
+        return body.isEmpty() ? typeName : typeName + "(" + body + ")";
+    }
+
+    private static ASTNode getInsertPosition(ElmFile sourceFile, String moduleName) {
```

Comment:
There is a concept of a code formatter in IntelliJ: http://www.jetbrains.org/intellij/sdk/docs/reference_guide/custom_language_support/code_formatting.html

It never was a top priority for me to implement it because I used `elm-format` successfully.